src/libraries/System.Private.CoreLib/src/System/AppContext.cs

```diff
@@ -135,6 +135,39 @@ public static void SetSwitch(string switchName, bool isEnabled)
             }
         }
 
+        internal static void LogSwitchValues(RuntimeEventSource ev)
+        {
+            if (s_switches is null)
+            {
+                return;
+            }
+
+            lock (s_switches)
+            {
+                foreach (var (k, v) in s_switches)
+                {
+                    // Convert bool to int because it's cheaper to log (no boxing)
+                    ev.LogAppContextSwitch(k, v ? 0 : 1);
+                }
+            }
+
+            if (s_dataStore is null)
+            {
+                return;
+            }
+
+            lock (s_dataStore)
+            {
+                foreach (var (k, v) in s_dataStore)
+                {
+                    if (v is bool b)
```

Comment:
This should use the same logic as `TryGetSwitch` - look for `if (GetData(switchName) is string value && bool.TryParse(value, out isEnabled))`